src/main/java/net/neoforged/neoforge/event/GrindstoneEvent.java

```diff
@@ -115,8 +118,22 @@ public static class OnTakeItem extends GrindstoneEvent implements ICancellableEv
         private ItemStack newTop = ItemStack.EMPTY;
         private ItemStack newBottom = ItemStack.EMPTY;
 
-        public OnTakeItem(ItemStack top, ItemStack bottom, int xp) {
+        private final ContainerLevelAccess access;
+        @Nullable
+        private final Player player;
+
+        public OnTakeItem(ContainerLevelAccess access, @Nullable Player player, ItemStack top, ItemStack bottom, int xp) {
             super(top, bottom, xp);
+            this.access = access;
+            this.player = player;
+        }
+
+        /**
+         * @deprecated Use {@link #OnTakeItem(ContainerLevelAccess, Player, ItemStack, ItemStack, int) the context-aware version} instead
+         */
+        @Deprecated(forRemoval = true, since = "1.21.8")
```

Comment:
Can you also add a to-do to remove the nullability annotation of the player when we remove the deprecated methods? Or should we keep it in case a mod wants to simulate a grindstone without a player (though it could use a fake player)? @XFactHD thoughts?

Reply:
I would be in favor if removing it.

Reply:
all set, added one to the CommonHooks method too